src/main/java/org/spongepowered/common/data/util/NbtDataUtil.java

```diff
@@ -116,42 +69,132 @@ private NbtDataUtil() {
+        public static final String CUSTOM_MANIPULATOR_TAG_LIST = "CustomManipulators";
+
+        public static final class Player {
+
+        }
+    }
+
+    public static final class Entity {
+
+        public static final String SPONGE_ENTITY_CREATOR = "Creator";
+        public static final String SPONGE_ENTITY_NOTIFIER = "Notifier";
+        public static final String CAN_GRIEF = "CanGrief";
+        public static final String ENTITY_TYPE_ID = "id";
+        public static final String ENTITY_POSITION = "Pos";
+        public static final String ENTITY_ROTATION = "Rotation";
+
+        public static final class Boat {
+
+            public static final String MAX_SPEED = "maxSpeed";
+            public static final String MOVE_ON_LAND = "moveOnLand";
+            public static final String OCCUPIED_DECELERATION_SPEED = "occupiedDecelerationSpeed";
+            public static final String UNOCCUPIED_DECELERATION_SPEED = "unoccupiedDecelerationSpeed";
+        }
+
+        public static final class Projectile {
+
+            public static final String DAMAGE_AMOUNT = "damageAmount";
+        }
+
+        public static final class Minecart {
+
+            public static final String MINECART_TYPE = "Type";
+        }
+    }
+
+    public static final class TileEntity {
+
+        public static final String BANNER_PATTERN_ID = "Pattern";
+        public static final String BANNER_PATTERN_COLOR = "Color";
+        // These are the various tag compound id's for getting to various places
+        public static final String TILE_ENTITY_ROOT = "BlockEntityTag";
+        public static final String TILE_ENTITY_ID = "id";
+        public static final String SIGN = "Sign";
+        public static final String POS_X = "x";
+        public static final String POS_Y = "y";
+        public static final String POS_Z = "z";
+    }
+
+    public static final class Compatibility {
+
+        public static final String USER_SPAWN_X = "SpawnX";
+        public static final String USER_SPAWN_Y = "SpawnY";
+        public static final String USER_SPAWN_Z = "SpawnZ";
+        public static final String USER_SPAWN_FORCED = "SpawnForced";
+        public static final String USER_SPAWN_LIST = "Spawns";
+        public static final String USER_SPAWN_DIM = "Dim";
+
+        public static final class Bukkit {
+
+            // Legacy migration tags from Bukkit
+            public static final String BUKKIT_ROOT = "bukkit";
+            public static final String FIRST_PLAYED = "firstPlayed";
+            public static final String LAST_PLAYED = "lastPlayed";
+        }
+
+        public static final class Forge {
+
+            // Compatibility tags for Forge
+            public static final String FORGE_ROOT = "ForgeData";
+            public static final String FORGE_ENTITY_TYPE = "entity_name";
+        }
+
+    }
+
+    public static final class World {
+
+        public static final String DIMENSION_TYPE = "dimensionType";
```

Comment:
I was considering that these would actually go under a new class for `Sponge` such that we can understand a difference between `Vanilla` and `Sponge` added constants.